Arrowgene.Ddon.GameServer/Handler/CraftStartQualityUpHandler.cs

```diff
@@ -0,0 +1,187 @@
+#nullable enable
+using System.Linq;
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Network;
+using Arrowgene.Logging;
+using System;
+using System.Collections.Generic;
+using System.Data.SqlTypes;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class CraftStartQualityUpHandler : GameRequestPacketHandler<C2SCraftStartQualityUpReq, S2CCraftStartQualityUpRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(CraftStartQualityUpHandler));
+        private static readonly List<StorageType> StorageEquipNBox = new List<StorageType> {
+            StorageType.ItemBagEquipment, StorageType.StorageBoxNormal, StorageType.StorageBoxExpansion,
+            StorageType.StorageChest, StorageType.CharacterEquipment, StorageType.PawnEquipment
+        };
+
+        private readonly ItemManager _itemManager;
+        private readonly Random _random;
```

Comment:
Same comments for this file as `Arrowgene.Ddon.GameServer/Handler/CraftStartEquipGradeUpHandler.cs`.